mlir/test/python/lib/CMakeLists.txt

```diff
@@ -26,8 +26,11 @@ add_mlir_public_c_api_library(MLIRCAPIPythonTestDialect
   MLIRPythonTestIncGen
 
   LINK_LIBS PUBLIC
-  MLIRCAPIInterfaces
-  MLIRCAPIIR
```

Comment:
i lost the thread - why is this necessary? something having to do with https://github.com/llvm/llvm-project/blob/ce5ac91a3b0fc3cdf6fe9559c817d49df99e4263/mlir/cmake/modules/AddMLIR.cmake#L420-L422 ?

Reply:
See:
```
mlir_c_target_link_libraries(MLIRCAPIPythonTestDialect
  PUBLIC
  MLIRCAPIInterfaces
  MLIRCAPIIR
)
```
See as well https://github.com/llvm/llvm-project/pull/119408 for rationale
and https://github.com/llvm/llvm-project/blob/ce5ac91a3b0fc3cdf6fe9559c817d49df99e4263/mlir/cmake/modules/AddMLIR.cmake#L784-L802

Also, I'd argue test libraries never cause a breaking change. They are not for production.

Reply:
what i mean by breaking change is you now have to use `mlir_c_target_link_libraries` if you do `MLIR_BUILD_MLIR_C_DYLIB=1` otherwise strange things will happen (like whatever you mean by "polluting" up above). maybe it's not a breaking change because prior if you did `MLIR_BUILD_MLIR_C_DYLIB=1` (and prior `mlir_c_target_link_libraries` added here) nothing at all would happen but at least things would break break.